src/vehicle.cpp

```diff
@@ -1558,97 +1565,23 @@ bool vehicle::can_mount(int const dx, int const dy, const vpart_id &id) const
         }
     }
 
-    //Seatbelts must be installed on a seat
-    if(part.has_flag("SEATBELT")) {
-        bool anchor_found = false;
-        for( const auto &elem : parts_in_square ) {
-            if( part_info( elem ).has_flag( "BELTABLE" ) ) {
-                anchor_found = true;
-            }
-        }
-        if(!anchor_found) {
-            return false;
-        }
-    }
-
-    //Internal must be installed into a cargo area.
-    if(part.has_flag("INTERNAL")) {
-        bool anchor_found = false;
-        for( const auto &elem : parts_in_square ) {
-            if( part_info( elem ).has_flag( "CARGO" ) ) {
-                anchor_found = true;
-            }
-        }
-        if(!anchor_found) {
-            return false;
-        }
-    }
-
-    // curtains must be installed on (reinforced)windshields
-    // TODO: do this automatically using "location":"on_mountpoint"
-    if (part.has_flag("CURTAIN")) {
-        bool anchor_found = false;
-        for( const auto &elem : parts_in_square ) {
-            if( part_info( elem ).has_flag( "WINDOW" ) ) {
-                anchor_found = true;
-            }
-        }
-        if (!anchor_found) {
-            return false;
-        }
-    }
-
-    // Security system must be installed on controls
-    if(part.has_flag("ON_CONTROLS")) {
-        bool anchor_found = false;
-        for( std::vector<int>::const_iterator it = parts_in_square.begin();
-             it != parts_in_square.end(); ++it ) {
-            if(part_info(*it).has_flag("CONTROLS")) {
-                anchor_found = true;
-            }
-        }
-        if(!anchor_found) {
-            return false;
-        }
-    }
-
-    // Cargo locks must go on lockable cargo containers
-    // TODO: do this automatically using "location":"on_mountpoint"
-    if(part.has_flag("CARGO_LOCKING")) {
-        bool anchor_found = false;
-        for( std::vector<int>::const_iterator it = parts_in_square.begin();
-             it != parts_in_square.end(); ++it ) {
-            if(part_info(*it).has_flag("LOCKABLE_CARGO")) {
-                anchor_found = true;
-            }
-        }
-        if(!anchor_found) {
-            return false;
+    if(!part.location.empty() && part.location.compare(0,3,"on_") == 0) {
+        std::string flag = part.location.substr(3); // masking off the 'on_'
```

Comment:
Not like this. String hacks are horrible to maintain and debug. The only place for string hacks where they are tolerable (but not good) is json loading itself, as long as the results are saved somewhere so that you can see them without breaking into the function directly.

You want to have `locations_provided` field on parts which provide locations and strings stored there should be exactly, 1-1, with locations required. No string transforms.

For backwards compatibility with mods, you could generate locations from a set of hardcoded flags (those that are used now) if the json does not explicitly provide locations.

Reply:
I have never seen a `locations_provided` field in a vehicle part. Would that act like a seat would have `"locations_provided": "seatbelt"` and then the seatbelt's location would be `"location": "seatbelt"`?

I agree that string hacking is a pain to debug but I didn't know how else to make this functionality.